pkg/infra/teams/team3/messaging.go

```diff
@@ -1,38 +1,68 @@
 package team3
 
 import (
+	"fmt"
 	"infra/game/agent"
 	"infra/game/commons"
 	"infra/game/message"
+	"sync"
 )
 
 // This is where you must compile your trust message. My example implementation takes ALL agents from the agent map **
 func (a *AgentThree) CompileTrustMessage(agentMap map[commons.ID]agent.Agent) message.Trust {
+	fmt.Println("AGENT 3 COMPOSED: message.Trust")
 
-	// fmt.Println("AGENT 3 COMPOSED: message.Trust")
-
-	keys := make([]commons.ID, len(agentMap))
+	//faireness = the function ids --> reputation number: which is the gossip
+	//send to everyone
+	keys := make([]commons.ID, len(agentMap)+len(a.TSN))
 
 	// ** it extracts the keys (i.e., the IDs) **
 	i := 0
-	for k := range agentMap {
+	for _, k := range a.TSN {
 		keys[i] = k
 		i++
 	}
 
+	for k := range agentMap {
+		keys[i] = k
+		i++
+	}
+	//fmt.Println("print keys: ", keys[:1])
 	// declare new trust message
 	trustMsg := new(message.Trust)
 
 	// ** and puts stuff inside
-	trustMsg.MakeNewTrust(keys[:1], make(map[string]int))
+	//trustMsg.MakeNewTrust(keys[:1], make(map[string]int))
+	num := int(a.samplePercent * float64(len(agentMap)))
+	trustMsg.MakeNewTrust(keys[0:num], a.reputationMap) //change the :1
```

Comment:
```suggestion
	trustMsg.MakeNewTrust(keys[:num], a.reputationMap)
```

Reply:
Is there not a social capital approach to how we choose to disseminate opinion? Why are we taking a random sample? Also, since we only slice linearly from index 0 -> `num`, the agents at the end of the list never get informed. I think this should be reworked

Reply:
Maps are iterated randomly in go, i.e. we always end up getting a different sample. Social capital is being used in the TSN. Order is: 1) Check TSN and THEN 2) fill remaining messages with random agents